### DISCUSSION TITLE
打包、编译、部署上线的问题

### DISCUSSION THREAD

**Discussion by CloudWoR** [nestcn/docs.nestjs.cn#265] [2-Q&A]
nest build时，依赖没有跟着一起打包，仅仅生成了一个main.js文件和tsconfig.app.tsbuildinfo文件，如果要部署到单机环境，还要把node_modules拷贝一份，不方便呀，有没有好的解决办法呢？

**Comment by zuohuadong** [comment]
提供几种方案供参考：
## 打包
可以使用 ncc/pkg 或者  nexe，可能需要改部分代码
https://github.com/vercel/ncc
https://github.com/nexe/nexe


## CI/CD
借鉴 git 自动部署，可以用caddy 实现自动从 github/gitee 拉取代码、安装、启动。
也可以使用 docker 做持续集成
